src/actions/helpers/index.ts

```diff
@@ -42,3 +60,38 @@ export const validateOwnerAndStatus = async (
   await validateUserOwnsDomain(domainId, potentialOwner, registrar);
   await validateStatus(domainId, registrar, desiredLock, potentialOwner);
 };
+
+/**
+ * Get a token's price using the CoinGecko API.
+ * If it can't be found on CoinGecko, that token's derived ETH ratio
+ * given by a DEX subgraph is used instead
+ * @param paymentTokenAddress The contract address of the ERC20 payment token
+ * @param derivedETH The derived ETH ratio
+ * @returns The price of the token in USD
+ */
+export const getTokenPrice = async (
```

Comment:
Same concept as before, can we abstract this?

Such as making another function called getCoinGeckoTokenPrice, that this method would then call. That would allow the coingecko specific code to exist there, while preserving normal pricing code here

Reply:
Makes sense, added